lib/internal/test_runner/harness.js

```diff
@@ -285,6 +325,11 @@ function lazyBootstrapRoot() {
 
 async function startSubtestAfterBootstrap(subtest) {
   if (subtest.root.harness.buildPromise) {
+    if (subtest.root.harness.globalSetupPromise) {
```

Comment:
This also feels off to me. I would expect the global setup to be part of the bootstrap process since it happens before any tests can run.